packages/agreement-email-sender/src/services/agreementEmailSenderService.ts

```diff
@@ -3,34 +3,41 @@ import path from "path";
 import fs from "fs/promises";
 import {
   AgreementV2,
+  Tenant,
+  TenantMail,
   fromAgreementV2,
-  genericInternalError,
+  tenantMailKind,
 } from "pagopa-interop-models";
 import {
   EmailManager,
-  Logger,
   buildHTMLTemplateService,
   dateAtRomeZone,
 } from "pagopa-interop-commons";
 import {
-  InstitutionResponse,
-  SelfcareV2Client,
-  mapInstitutionError,
-} from "pagopa-interop-selfcare-v2-client";
-import {
+  activationDateNotFound,
   descriptorNotFound,
   eServiceNotFound,
-  institutionNotFound,
-  selfcareIdNotFound,
+  tenantDigitalAddressNotFound,
+  tenantNotFound,
 } from "../models/errors.js";
 import { agreementEmailSenderConfig } from "../utilities/config.js";
 import { ReadModelService } from "./readModelService.js";
 
+export const retrieveTenantDigitalAddress = async (
+  tenant: Tenant
+): Promise<TenantMail> => {
```

Comment:
the function does not perform async operations

Reply:
Right! Fixed